src/index.js

```diff
@@ -208,93 +274,83 @@ class ExtractTextPlugin {
                         );
                       }
                       callback();
-                    });
-                  } else {
-                    if (meta.content) {
-                      extractCompilation.addResultToChunk(
-                        module.identifier(),
-                        meta.content,
-                        module,
-                        extractedChunk
-                      );
                     }
-                    callback();
+                  } else callback();
+                },
+                (err) => {
```

Comment:
Can't you directly use `callback` here?

Reply:
Reverting this change, because it broke the whole plugin 😨
This whole section is a jumbled up mess of asynchrony.
Nevertheless, I'm wholly behind the idea of refactoring, but first I'd like to see the core functionality fixed.